src/component/video-card/VideoCard.jsx

```diff
@@ -4,82 +4,123 @@ import WatchLaterOutlinedIcon from '@mui/icons-material/WatchLaterOutlined';
 import PlayCircleOutlineOutlinedIcon from '@mui/icons-material/PlayCircleOutlineOutlined';
 import DeleteOutlineOutlinedIcon from '@mui/icons-material/DeleteOutlineOutlined';
 
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useLocation } from 'react-router-dom';
 import { useState } from 'react';
 
-import { addItemToWatchLater, removeItemFromWatchLater } from 'service';
+import { addItemToWatchLater, removeItemFromWatchLater, addVideoToHistory, removeVideoFromHistory } from 'service';
 import { useData, useAuth } from 'context';
 import { useToast } from 'custom-hook/useToast';
 
-const VideoCard = ({video}) => {
+const VideoCard = ({ video }) => {
   const navigate = useNavigate();
-  const { videoDispatch, videoState:{watchlater}} = useData();
-  const { auth:{isAuth} } = useAuth();
+  const location = useLocation();
+  const { videoDispatch, videoState: { watchlater, history } } = useData();
+  const { auth: { isAuth } } = useAuth();
```

Comment:
you can destructure can pass the values from the context itself so that you don't have to destructure it whenever you use it

Reply:
Yes didn't think of this in flow-flow. Will refactor